### DISCUSSION TITLE
Unintuitive plugin ordering in packer_compiled.lua

### DISCUSSION THREAD

**Discussion by Kirens** [wbthomason/packer.nvim#1130] [Q&A]
As far as I understand, the plugin ordering of `use { ... }` invocations is not preserved for config loading in _packer_compiled.lua_. This seems to be contrary to intuitive assumptions. (E.g. as seen in this popular nvim-config: https://github.com/jdhao/nvim-config/blob/8b10be60d45f2505d1f6962014640a123b814614/lua/plugins.lua#L46-L47)

Is this correct and would it be reasonable to preserve the ordering? Or should this just be clarified in the documentation?

**Comment by yutkat** [comment]
It is true that plugin managers that use `packadd` will have an undefined loading order. If you are worried about it, you can use something that does not use `packadd`, such as vim-plug.

If you want to use packer, you can use `after` to define the startup order, so I use this feature.
https://github.com/wbthomason/packer.nvim/blob/6afb67460283f0e990d35d229fd38fdc04063e0a/doc/packer.txt#L586

**Reply by Kirens** [reply]
I suppose that might make it an unfair assumption. Although, I was specifically thinking of the plugins that are not optional. In that case the config seems to be in‑lined in the order that `pairs` iterate through the table.

https://github.com/wbthomason/packer.nvim/blob/6afb67460283f0e990d35d229fd38fdc04063e0a/lua/packer/compile.lua#L530-L534

But I guess the configs are meant to be pretty independent, in general.

**Reply by yutkat** [reply]
I don't know about impatient.nvim, but if you absolutely want to load it first, you could load it manually first, like packer.

```lua
vim.cmd("packadd impatient.nvim")
require('impatient')

...
packer.startup {
  function(use)
    use { "lewis6991/impatient.nvim", opt = true }
```
